### ISSUE TITLE
Error Accessing Model Information

### ISSUE THREAD

**Issue by aabest** [ModelSEED/Model-SEED-core#163]
When using the mdl-printcytoseed command, we are not getting correct model information printed to the appropriate cytoseed files.

The program is returning no information or an error in the mdl-printcytoseed subroutine in ModelDriver.pm when calling "get_model_data" and "biomass_reaction_details" in the following code:

```
open(FH, ">".$cmdir."/model_data") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
my $md = $fbaObj->get_model_data({ "id" => [$args->{model}] });
print FH $dumper->dump($md->{$args->{model}});
close FH;
print "Model data printed...\n";
open(FH, ">".$cmdir."/biomass_reaction_details") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
print FH $dumper->dump($fbaObj->get_biomass_reaction_data({ "model" => [$args->{model}] }));
close FH;
print "Biomass data printed...\n";
```

Output for model data is:

--- ~

When the biomass is accessed, it returns:

---

error:
- iYusMp.906 either nonexistant or nonaccessible

This error is being generated from code in FBAMODEL.pm

Accessing other data in the model appears to work (for instance, compound data, reaction data).

Additionally, accessing the compound data seems to be taking about 20 minutes to process this call:

my $cids = $fbaObj->get_compound_id_list({ "id" => [$args->{model}] });
    open(FH, ">".$cmdir."/compound_details") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
    my $cpds = $fbaObj->get_compound_data({ "id" => $cids->{$args->{model}} });
    print FH $dumper->dump($cpds);
    close FH;
    print "Compound data printed...\n";

Seems like that is very long, especially for a small genome/model.

**Comment by aabest** [comment]
An update... For some reason, two nights ago I was able to successfully print the files from the mdl-printcytoseed command from my personal laptop installation (information in the original post above was coming from a different installation).

Yesterday, I performed an ms-update, and now I am unable to use this command on my laptop. The output is exactly as described above... it was working, model existed, nothing changed except the ms-update... I was simply testing to see that I was still able to successfully print the files for cytoseed.

any ideas?

**Comment by mdejongh** [comment]
This probably has to do with access rights. Are you logged in to
ModelSEED as the user who loaded the model?

On Mon, Jun 18, 2012 at 4:46 PM, aabest
reply@reply.github.com
wrote:

> When using the mdl-printcytoseed command, we are not getting correct model information printed to the appropriate cytoseed files.
> 
> The program is returning no information or an error in the mdl-printcytoseed subroutine in ModelDriver.pm when calling "get_model_data" and "biomass_reaction_details" in the following code:
> 
>        open(FH, ">".$cmdir."/model_data") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
>        my $md = $fbaObj->get_model_data({ "id" => [$args->{model}] });
>        print FH $dumper->dump($md->{$args->{model}});
>        close FH;
>        print "Model data printed...\n";
>        open(FH, ">".$cmdir."/biomass_reaction_details") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
>        print FH $dumper->dump($fbaObj->get_biomass_reaction_data({ "model" => [$args->{model}] }));
>        close FH;
>        print "Biomass data printed...\n";
> 
> Output for model data is:
> 
> --- ~
> 
> When the biomass is accessed, it returns:
> 
> ---
> 
> error:
>  - iYusMp.906 either nonexistant or nonaccessible
> 
> This error is being generated from code in FBAMODEL.pm
> 
> Accessing other data in the model appears to work (for instance, compound data, reaction data).
> 
> Additionally, accessing the compound data seems to be taking about 20 minutes to process this call:
> 
> my $cids = $fbaObj->get_compound_id_list({ "id" => [$args->{model}] });
>        open(FH, ">".$cmdir."/compound_details") or ModelSEED::utilities::ERROR("Could not open file: $!\n");
>        my $cpds = $fbaObj->get_compound_data({ "id" => $cids->{$args->{model}} });
>        print FH $dumper->dump($cpds);
>        close FH;
>        print "Compound data printed...\n";
> 
> Seems like that is very long, especially for a small genome/model.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ModelSEED/Model-SEED-core/issues/163

## 

Matt DeJongh, Ph.D.
Computer Science Department
Hope College, Holland, MI 49423
(616) 395-7429

**Comment by aabest** [comment]
I made the model on my local installation of Model-SEED, gapfilled on specific media, performed a checkgrowth, performed fva, and then printed to cytoseed successfully. Then the next day... it did not work on the same installation. I did not log out or change users during the interim. I did run an ms-update to the latest version of Model-SEED between the trials.

**Comment by mdejongh** [comment]
For now, try commenting out as follows (the two lines with +# at the beginning):

+++ b/lib/ModelSEED/FIGMODEL/FIGMODELdatabase.pm
@@ -299,9 +299,9 @@ sub get_objects {
                my $finalObjs;
                for (my $i=0; $i < @{$objs}; $i++) {
             my $rights = $self->get_object_rights($objs->[$i], $type);
-            if(keys %$rights > 0) {
  +#            if(keys %$rights > 0) {
                              push(@{$finalObjs},
  ModelSEED::FIGMODEL::FIGMODELdata->new($objs->[$i], $self->fig
-                       }
  +#                      }
              }
              $objs = $finalObjs;
      }

On Mon, Jun 25, 2012 at 11:23 AM, aabest
reply@reply.github.com
wrote:

> I made the model on my local installation of Model-SEED, gapfilled on specific media, performed a checkgrowth, performed fva, and then printed to cytoseed successfully. Then the next day... it did not work on the same installation. I did not log out or change users during the interim. I did run an ms-update to the latest version of Model-SEED between the trials.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ModelSEED/Model-SEED-core/issues/163#issuecomment-6550352

## 

Matt DeJongh, Ph.D.
Computer Science Department
Hope College, Holland, MI 49423
(616) 395-7429

**Comment by aabest** [comment]
Okay, this works with the above change. I will change on the other installations, as well.

Is there an underlying issue?

Thanks!

**Comment by mdejongh** [comment]
The underlying issue is problem with access rights - I'm hoping that
Chris et al will fix it, but for now the commenting out seems to work.

Are you and your students able to print cytoseed files now?

On Mon, Jun 25, 2012 at 2:48 PM, aabest
reply@reply.github.com
wrote:

> Okay, this works with the above change. I will change on the other installations, as well.
> 
> Is there an underlying issue?
> 
> Thanks!
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/ModelSEED/Model-SEED-core/issues/163#issuecomment-6556152

## 

Matt DeJongh, Ph.D.
Computer Science Department
Hope College, Holland, MI 49423
(616) 395-7429

**Comment by aabest** [comment]
Yes, we can print the cytoseed files successfully. Thanks for getting a workaround together.
